### PR TITLE
chore: Add FOSSA check to GH actions

### PR DESCRIPTION
**Pull request by ghislainbourgeois**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.14.4'
```
Comment (gab-arrobo) [inline .github/workflows/main.yml:23]:
```suggestion
        go-version-file: 'go.mod'
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
Comment (gab-arrobo) [inline .github/workflows/main.yml:18]:
```suggestion
    - uses: actions/checkout@v4
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
```
Comment (gab-arrobo) [inline .github/workflows/main.yml:21]:
```suggestion
      uses: actions/setup-go@v5
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      with:
+        go-version: '1.14.4'
+
+    - name: Build
+      run: go build -v ./...
+
+    - name: Test
+      run: go test -v ./...
```
Comment (gab-arrobo) [inline .github/workflows/main.yml:29]: Should not this be deleted given that you already have a `build` and `unit-tests` actions below?

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Build
+        run: go build
```
Comment (gab-arrobo) [inline .github/workflows/main.yml:41]:
```suggestion
        run: go build ./...
```

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+  fossa-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: FOSSA scan
+        uses: fossa-contrib/fossa-action@v3
+        with:
+          fossa-api-key: 9dc8fa92e3dd565687317beb87b56d89
```
Comment (gab-arrobo) [inline .github/workflows/main.yml:59]: Should we move these actions to the end of the file such we try to keep as much consistency as possible with the GHA file in the other repos?

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Main workflow
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```
Comment (gab-arrobo) [inline .github/workflows/main.yml:11]: To make it similar to the GHA file in the other repos
```suggestion
  pull_request:
    branches:
      - main
  push:
    branches:
      - main
```

**Review by gab-arrobo — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 4 above)
Comment (ghislainbourgeois) [reply .github/workflows/main.yml]: I could swear I had removed those, but seems not! I deleted that duplicate section.

File: .github/workflows/main.yml (hunk 5 above)
Comment (ghislainbourgeois) [reply .github/workflows/main.yml]: Done

File: .github/workflows/main.yml (hunk 6 above)
Comment (ghislainbourgeois) [reply .github/workflows/main.yml:56]: Done

File: .github/workflows/main.yml (hunk 7 above)
Comment (ghislainbourgeois) [reply .github/workflows/main.yml]: Done

**Review by gab-arrobo — APPROVED** [review]
+1

**Comment by gab-arrobo** [comment]
@ghislainbourgeois there is license issue with one file (Namf_Communication/.openapi-generator/VERSION). How should we fixed it?

**Comment by ghislainbourgeois** [comment]
> @ghislainbourgeois there is license issue with one file (Namf_Communication/.openapi-generator/VERSION). How should we fixed it?

I removed the file, as it seems unused and it is only present in that specific directory.

**Review by gab-arrobo — APPROVED** [review]
+1

**Pull request merged**